### PR TITLE
feat: replace rest_framework_swagger with drf_yasg

### PR DESCRIPTION
**Pull request by NeOneSoft**
This PR was created following edx/upgrades#17 to switch django rest swagger to drf-yasg

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @NeOneSoft! I've created [OSPR-6284](https://openedx.atlassian.net/browse/OSPR-6284) to keep track of it in JIRA, where we prioritize reviews. Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

File: credentials/urls.py [hunk 1]
```diff
@@ -34,11 +36,21 @@
...
+   public=False,
+   permission_classes=[permissions.AllowAny],
+)
+
 urlpatterns = oauth2_urlpatterns + [
     re_path(r"^admin/", admin.site.urls),
     re_path(r"^api/", include(("credentials.apps.api.urls", "api"), namespace="api")),
     re_path(r"^api-auth/", include((oauth2_urlpatterns, "rest_framework"), namespace="rest_framework")),
-    re_path(r"^api-docs/", get_swagger_view(title="Credentials API"), name="api_docs"),
+    re_path(r"^api-docs/$", schema_view.with_ui('swagger', cache_timeout=0), name="api_docs"),,
```
Comment (justinhynes) [inline credentials/urls.py:53]: @NeOneSoft Looks like we may have an extra comma here that is causing issues the config files.

Comment (NeOneSoft) [reply credentials/urls.py]: @justinhynes Thank you so much!!!

**Review by justinhynes — CHANGES_REQUESTED** [review]
@NeOneSoft I think we're really close getting this PR wrapped up. It looks like we're running into a few quality issues. We should be able to use our existing tools to automate the fixes. Locally you can try running:
`make isort` - this should fix the error with the imports in the wrong order, then
`black .` - (note the period) black is a code formatter used in this repo to keep syntax common across all files.

From there you can run `make quality` locally to have isort/black/pylint run to point out any other potential quality issues.

I think after this we should be good to squash and merge the changes. :)

**Comment by NeOneSoft** [comment]
@justinhynes !!!!Thank you so much for you instructions and support. I already apply the suggested commands and once our quality issues were checked  I will be doing squash!!

**Comment by NeOneSoft** [comment]
Hey@justinhynes!! after apply isort, black and quality we're still having and import issue on copy_catalog.py file but I think that the sort is correct.

**Comment by justinhynes** [comment]
@NeOneSoft I'm sorry this is such a pain in the neck! I definitely didn't expect 126 files to be updated. I'm curious as to what is going on here. Going to take some extra time to think this through... thanks for your patience.

**Comment by NeOneSoft** [comment]
No problem @justinhynes!!!!I'll be attending your comments, thanks again for all your support!!

**Comment by justinhynes** [comment]
@NeOneSoft Hey there! Apologies that I haven't been able to return and triage your issues any further. I am wrapping up my day here soon and actually am off until January. If it's okay with you, we can try and get this sorted out shortly after the new year. Thanks for understanding and talk soon!

**Comment by NeOneSoft** [comment]
Thank you so much @justinhynes, enjoy and happy holidays!!!!

**Comment by natabene** [comment]
@NeOneSoft Thanks so much for your contribution. @justinhynes Thank you for reviewing!

**Comment by hurtstotouchfire** [comment]
Hey @NeOneSoft, I'd like to get this merged but I think we will need to reduce the number of touched files. Are you able to rebase and do some commit cleanup? I would like to see 5f2d5caa630dc308c476b1f9a9ff60b0b4ead7d0 fixed up to the commit before it (just in case we end up dividing up this PR, I don't want that fixup to be separated from the commit it fixes) and then pull out 93db6794b1ebc444a893ae2cdbd59622b953dcee for the time being. You could stash it in another branch for now, and then we can determine if we want to pull some of it in piecemeal. Does that sound doable?

**Comment by NeOneSoft** [comment]
> Hey @NeOneSoft, I'd like to get this merged but I think we will need to reduce the number of touched files. Are you able to rebase and do some commit cleanup? I would like to see [5f2d5ca](https://github.com/openedx/credentials/commit/5f2d5caa630dc308c476b1f9a9ff60b0b4ead7d0) fixed up to the commit before it (just in case we end up dividing up this PR, I don't want that fixup to be separated from the commit it fixes) and then pull out [93db679](https://github.com/openedx/credentials/commit/93db6794b1ebc444a893ae2cdbd59622b953dcee) for the time being. You could stash it in another branch for now, and then we can determine if we want to pull some of it in piecemeal. Does that sound doable?

**Comment by natabene** [comment]
@NeOneSoft Please let us know once you are ready for this to be reviewed again.

**Comment by NeOneSoft** [comment]
Hi @natabene of course, I'll be working on it this week!

**Comment by NeOneSoft** [comment]
Hi @hurtstotouchfire!!,Are you agree if I store the changes of [94db679](https://github.com/openedx/credentials/commit/93db6794b1ebc444a893ae2cdbd59622b953dcee)in another branch as you suggested and then I reset and squash to [5f25ca](https://github.com/openedx/credentials/commit/5f2d5caa630dc308c476b1f9a9ff60b0b4ead7d0)?Thanks in advance

**Review by hurtstotouchfire — APPROVED** [review]
As long as @justinhynes has no blocking concerns I think this is good to go from our perspective. I think you may need to resolve conflicts though.

**Comment by NeOneSoft** [comment]
Hi @justinhynes !!Thanks a lot for your review. I already run the commands are you suggested. I'll be glad to attend your comments and thanks again

**Comment by justinhynes** [comment]
@NeOneSoft Great, thank you for those changes. I took a look and it looks good to go. I'm going to approve the PR.

**Review by justinhynes — APPROVED** [review]

**Comment by hurtstotouchfire** [comment]
Hmmmm we have merge conflicts again. Sorry @NeOneSoft this tends to happen when there are dependency updates because we are also merging update PRs so they tend to conflict.

**Comment by NeOneSoft** [comment]
No worries @hurtstotouchfire !!!, should we wait to get updated all our dependencies before apply make upgrade again?

**Comment by hurtstotouchfire** [comment]
@NeOneSoft I had one more update PR to merge which may or may not conflict for you but after this you should be good to go.

This is the PR: https://github.com/openedx/credentials/pull/1689

Let us know when you're de-conflicted!

**Comment by NeOneSoft** [comment]
Hi @hurtstotouchfire!!, I had errors rebasing this branch so I created a new one on [PR](https://github.com/openedx/credentials/pull/1690) with exactly the same changes but starting from updated master. I think that this should be solve the conflicts. For last, could you please review it  to be sure that everything it's ok?. For know titled as "draft" but I will change it once this is ready for merge. Thanks a lot !!

**Comment by NeOneSoft** [comment]
This PR can be closed on behalf of [PR](https://github.com/openedx/credentials/pull/1690) where the required changes were done and some issues regarding python 3.8 dependencies where solved. Thanks everyone for your all your support!!!!

**Pull request closed without merging**